### ISSUE TITLE
Writing lock file in cache hangs for non-transient OpErrors, such as PermissionDeniedError

### ISSUE THREAD

**Issue by twise2** [tensorflow/hub#415]
Expected Behavior: Fail or Error

Actual Behavior: Hangs atleast 24 hours.

If a user with higher permissions loads a tensorflow hub embedding and stores them in a cache, a user without permission to the cached store will hand when running the hub.Module(module_url) as the file will exists but they will not be able to read from it. 

Code to be run by admin then user without permissions to TFHUB_CACHE_DIR:

```
import tensorflow as tf
import tensorflow_hub as hub
module_url = "https://tfhub.dev/google/universal-sentence-encoder-large/3"
text_input = tf.placeholder(dtype=tf.string, shape=[None])
embed = hub.Module(module_url)
```

conda env of tensorflow environment:

```
name: ubuntu
channels:
  - pytorch
  - defaults
dependencies:
  - _libgcc_mutex=0.1
  - _tflow_select=2.3.0
  - absl-py=0.8.1
  - asn1crypto=1.2.0
  - astor=0.8.0
  - beautifulsoup4=4.8.1
  - blas=1.0
  - c-ares=1.15.0
  - ca-certificates=2019.10.16
  - certifi=2019.9.11
  - cffi=1.13.1
  - chardet=3.0.4
  - click=7.0
  - cryptography=2.8
  - faiss-cpu=1.6.0
  - flask=1.1.1
  - gast=0.3.2
  - grpcio=1.16.1
  - gunicorn=19.9.0
  - h5py=2.9.0
  - hdf5=1.10.4
  - idna=2.8
  - intel-openmp=2019.4
  - itsdangerous=1.1.0
  - jinja2=2.10.3
  - keras-applications=1.0.8
  - keras-preprocessing=1.1.0
  - libedit=3.1.20181209
  - libffi=3.2.1
  - libgcc-ng=9.1.0
  - libgfortran-ng=7.3.0
  - libprotobuf=3.9.2
  - libstdcxx-ng=9.1.0
  - markdown=3.1.1
  - markupsafe=1.1.1
  - mkl=2019.4
  - mkl-service=2.3.0
  - mkl_fft=1.0.15
  - mkl_random=1.1.0
  - mock=3.0.5
  - ncurses=6.1
  - numpy=1.17.3
  - numpy-base=1.17.3
  - openssl=1.1.1d
  - pip=19.3.1
  - protobuf=3.9.2
  - pycparser=2.19
  - pyopenssl=19.0.0
  - pysocks=1.7.1
  - python=3.7.5
  - readline=7.0
  - requests=2.22.0
  - scipy=1.3.1
  - setuptools=41.6.0
  - six=1.12.0
  - soupsieve=1.9.3
  - sqlite=3.30.1
  - tensorboard=1.13.1
  - tensorflow=1.13.1
  - tensorflow-base=1.13.1
  - tensorflow-estimator=1.13.0
  - tensorflow-hub=0.6.0
  - termcolor=1.1.0
  - tk=8.6.8
  - urllib3=1.24.2
  - werkzeug=0.16.0
  - wheel=0.33.6
  - xz=5.2.4
  - zlib=1.2.11
prefix: /home/miniconda/envs/ubuntu
```
executable by running ```conda env create -f <above code>```

**Comment by arnoegw** [comment]
Confirmed: atomic_download() retries indefinitely when creating the lock file fails with any OpError other than NotFoundError. Source location: https://github.com/tensorflow/hub/blob/r0.6/tensorflow_hub/resolver.py#L388

I think this should be fixed by failing for more types of non-transient errors besides NotFoundError, notably for the PermissionDeniedError likely seen in the reported instance of this problem.

**Comment by arnoegw** [comment]
@twise2, thanks a lot for reporting this. Commit https://github.com/tensorflow/hub/commit/ed61dbd3a042251a414e29bd4811a3a403c59a25 has fixed this on the master branch and will be included in the tensorflow-hub 0.8 release eventually.  If you need it sooner than that, please wait for the next tf-hub-nightly pip package and use that.
